### PR TITLE
Adds OpenSCAP 0.5.3 params

### PR DESCRIPTION
**Pull request by shlomizadok**
OpenSCAP introduces three new params: `contentdir`, `reportsdir` & `failed_dir`

### CODE REVIEW

File: templates/plugin/openscap.yml.erb [hunk 1]
```diff
@@ -5,5 +5,18 @@
...
 
 # Directory where OpenSCAP audits are stored
-# before they are forwarded to Foreman
+# if they failed to post to Foreman. smart_proxy_openscap_send will
+# to re-send them.
 :spooldir: <%= scope.lookupvar('::foreman_proxy::plugin::openscap::spooldir') %>
+
+# Directory where OpenSCAP content XML are stored
+# So we will not request the XML from Foreman each time
+:contentdir: /var/lib/openscap/content
```
Comment (ekohl) [inline templates/plugin/openscap.yml.erb]: You need to use the parameters here.

Comment (shlomizadok) [inline templates/plugin/openscap.yml.erb]: Yeah, sorry. fixed.

**Comment by ekohl** [comment]
:+1:

File: manifests/plugin/openscap.pp [hunk 2]
```diff
@@ -16,17 +16,29 @@
 #
 # $spooldir::                   Directory where OpenSCAP audits are stored
 #                               before they are forwarded to Foreman
+# $contentdir::                 Directory where OpenSCAP content XML are stored
+#                               So we will not request the XML from Foreman each time
+# $reportsdir::                 Directory where OpenSCAP report XML are stored
```
Comment (domcleal) [inline manifests/plugin/openscap.pp]: missing space between parameters

File: manifests/plugin/openscap/params.pp [hunk 3]
```diff
@@ -6,4 +6,7 @@
   $listen_on               = 'https'
   $openscap_send_log_file  = '/var/log/foreman-proxy/openscap-send.log'
   $spooldir                = '/var/spool/foreman-proxy/openscap'
+  $contentdir              = '/usr/share/foreman-proxy/openscap/content'
```
Comment (domcleal) [inline manifests/plugin/openscap/params.pp]: Not really a comment on this PR, but you should reconsider the location of these files, it sounds like an FHS violation.

**Comment by domcleal** [comment]
Tests will also need updating.

File: manifests/plugin/openscap.pp (hunk 2 above)
Comment (shlomizadok) [inline manifests/plugin/openscap.pp]: yeah will update as tests finish

File: manifests/plugin/openscap/params.pp (hunk 3 above)
Comment (shlomizadok) [inline manifests/plugin/openscap/params.pp]: Shall I convert them to relative path? (packaging creates symlinks when installing)

Comment (domcleal) [inline manifests/plugin/openscap/params.pp]: Ah, if they're already symlinks then perhaps point them to the target instead.

Comment (shlomizadok) [inline manifests/plugin/openscap/params.pp]: Not sure I understand your last comment.
When installing we create symlinks: `openscap/content`, `openscap/reports`, etc.
Shall I point to `openscap/content` or to the full path `/usr/share/foreman-proxy/openscap/content` ?

Comment (domcleal) [inline manifests/plugin/openscap/params.pp]: I'd have thought the full path at all times.  I suggested using the symlink target path though, not the symlink itself?

Comment (ekohl) [inline manifests/plugin/openscap/params.pp]: What is the absolute path they now resolve to?

Comment (shlomizadok) [inline manifests/plugin/openscap/params.pp]: `/usr/share/foreman-proxy/openscap/content`, `/usr/share/foreman-proxy/openscap/reports`, etc.
So I think this is right. Not sure why it is an FHS violation though.

Comment (ekohl) [inline manifests/plugin/openscap/params.pp]: It depends on what you do with it, but foreman-proxy should not write to ```/usr/share/foreman-proxy```. reports implies these are the uploaded reports from clients. You would expect them to be somewhere in ```/var``` (and ```/usr/share/foreman-proxy/openscap/reports``` could be a symlink.

Comment (shlomizadok) [inline manifests/plugin/openscap/params.pp]: this is embarrassing. It was `/var/lib/` all along. Updating
ref [#1](https://github.com/theforeman/foreman-packaging/blob/rpm/1.11/rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec#L6), [#2](https://github.com/theforeman/foreman-packaging/blob/rpm/1.11/rubygem-smart_proxy_openscap/rubygem-smart_proxy_openscap.spec#L88-L91)

**Comment by shlomizadok** [comment]
Updated paths & tests

**Comment by mmoll** [comment]
merged, thanks @shlomizadok!

**Pull request closed without merging**